src/convolution.cpp

```diff
@@ -338,18 +424,11 @@ ConvolutionDescriptor::WrwGetValidWorkSpaceSizeGemm(const TensorDescriptor& dyDe
     return 0;
 }
 
-std::size_t ConvolutionDescriptor::ForwardGetWorkSpaceSize(Handle& handle,
-                                                           const TensorDescriptor& wDesc,
-                                                           const TensorDescriptor& xDesc,
-                                                           const TensorDescriptor& yDesc) const
+std::size_t ConvolutionDescriptor::GetWorkSpaceSize(ExecutionContext ctx,
+                                                    const conv::ProblemDescription& problem) const
 {
     MIOPEN_LOG_I2("");
 
-    const auto problem = ProblemDescription{xDesc, wDesc, yDesc, *this, conv::Direction::Forward};
-    auto ctx           = ConvolutionContext{};
-    ctx.SetStream(&handle);
-    ctx.DetectRocm();
-    ctx.SetupFloats(problem);
```

Comment:
@DrizztDoUrden It seems that changes like this lead to [SWDEV-404537](https://ontrack-internal.amd.com/browse/SWDEV-404537), which is fixed in #2288. I am wondering if similar problems are still hiding somewhere in our code even after the fix.

As you can see, in the old code we do not fully construct ExecutionContext from handle and manually call DetectRocm when necessary. The refactored code is missing this. Shall we always call DetectRocm in ctor that accepts handle (`ExecutionContext(Handle* stream_)`) to prevent this problem? And #2288 can re reverted at once.

Oops, I see that #2289 is opened.

Reply:
@DrizztDoUrden BTW I am wondering if we still need the default ctor `ExecutionContext::ExecutionContext()`. If we can get rid of it, then I think we can make `DetectRocm()` the private member of `ExecutionContext` and then never call it manually.

Reply:
Yes, I think that would be a proper way to do this.

Reply:
[resolved]